src/scenedit/scen.main.cpp

```diff
@@ -609,39 +627,9 @@ void handle_menu_choice(eMenu item_hit) {
 	redraw_screen();
 }
 
-static void handleUpdateWhileScrolling(volatile bool& doneScrolling) {
-	while(!doneScrolling) {
-		sf::sleep(sf::milliseconds(10));
-		// TODO: redraw_screen should probably take the argument specifying what to update
-		redraw_screen(/*REFRESH_RIGHT_BAR*/);
-		if(overall_mode < MODE_MAIN_SCREEN || overall_mode == MODE_EDIT_TYPES)
-			set_up_terrain_buttons(false);
-	}
-	mainPtr.setActive(false);
-}
-
 void Mouse_Pressed(sf::Event const & event) {
-	
-	// Translate coordinates
 	location mousePos { translate_mouse_coordinates({event.mouseButton.x, event.mouseButton.y}) }; 
-	
-	volatile bool doneScrolling = false;
-	if(right_sbar->isVisible() && mousePos.in(right_sbar->getBounds())) {
-		mainPtr.setActive(false);
-		boost::thread updater(std::bind(handleUpdateWhileScrolling, std::ref(doneScrolling)));
-		right_sbar->handleClick(mousePos);
-		doneScrolling = true;
-		updater.join();
-		redraw_screen(/*REFRESH_RIGHT_BAR*/);
-	} else if(pal_sbar->isVisible() && mousePos.in(pal_sbar->getBounds())) {
-		mainPtr.setActive(false);
-		boost::thread updater(std::bind(handleUpdateWhileScrolling, std::ref(doneScrolling)));
-		pal_sbar->handleClick(mousePos);
-		doneScrolling = true;
-		updater.join();
-		redraw_screen(/*REFRESH_RIGHT_BAR*/);
-		set_up_terrain_buttons(false);
-	} else handle_action(mousePos,event);
+	handle_action(mousePos,event);
```

Comment:
Feels like there's no reason for the `Mouse_Pressed` function to even exist anymore...

Reply:
Coordinate translation is a long statement, I do not like having them inline, so I'll leave this as is.